### PR TITLE
added Google Cloud Storage (GCS) support

### LINKED ISSUE #38: [Feature Request] Support GCS private storage
Hello,

A part of my customers use other cloud vendors (eg. GCP, Azure, Alibaba), it would nice to support them in order to be cloud agnostic.

Regarding my roadmap, GCP is a top priority so I'm ready to contribute to create a _Google Cloud Storage_ internal module.

Afterwards, it would be interesting to define what should be possible to [configure](https://github.com/valentindeaconu/terralist/blob/master/example.yaml) for the end-user.

**Examples**:

1. possibility to define a global private storage -> apply to all API resources (modules, providers, etc.) [_which what is currently available_] and authorities
2. possibility to define a private storage at the authority level (eg. authority A uses AWS whereas authority B uses GCS)
3. possibility to define a private storage at the API resource level (eg. AWS for modules and GCS providers)

If you think that this needs is legitimate, I'm ready to exchange further on this topic ;)

### PR DESCRIPTION
**Pull request by alexbacchin**
Added support for GCS, including adding new configuration variables in the documentation

resolves #38

### CODE REVIEW

File: web/dist/.gitkeep
Comment (valentindeaconu) [inline web/dist/.gitkeep:1]: Can you please not delete this file?

**Review by valentindeaconu — CHANGES_REQUESTED** [review]
Thank you for this amazing contribution! 🚀 

I left some comments, can you please address them?

File: docs/configuration.md [hunk 1]
```diff
@@ -671,6 +671,55 @@ The number of minutes after which the Azure Shared Access Signature(SAS) should
...
+
+| Name | Value |
+| --- | --- |
+| type | int |
+| required | no |
+| default | `15` |
+| cli | `--gcs-sign-expire` |
+| env | `TERRALIST_GCS_SIGN_EXPIRE` |
+
+### `s3-access-key-id`
```
[the remark this answers is not in the document]
Comment (alexbacchin) [reply docs/configuration.md]: Done

File: pkg/storage/gcs/config.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	if os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") != "" && c.ServiceAccountCredFilePath == "" {
+		c.ServiceAccountCredFilePath = os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	}
+
+	if c.ServiceAccountCredFilePath != "" {
+		c.DefaultCredentials = false
+
+	} else {
+		c.DefaultCredentials = false
+	}
```
[the remark this answers is not in the document]
Comment (alexbacchin) [reply pkg/storage/gcs/config.go]: Thank you for catching that. Fixed

**Review by valentindeaconu — APPROVED** [review]

**Pull request merged**